File: 7-kyu/valid-parentheses/javascript/solution.js
const validParentheses = parenStr => {
  if (parenStr.length === 0) {
    return true;
  }
  const queue = [];
  let isValid = false;
  for (const ch of parenStr) {
    if (ch === "(") {
      queue.push(ch);
    } else if (ch === ")") {
      if (queue.length > 0 && queue[0] === "(") {
        isValid = true;
        queue.shift();
      } else {
        isValid = false;
        break;
      }
    }
  }
  return queue.length === 0 && isValid;
};

File: 7-kyu/valid-parentheses/swift/solution.swift
func validParentheses(_ str: String) -> Bool {
  var queue: [Character] = []
  var isValid = true
  for ch in str {
    if ch == "(" {
      queue.append(ch)
    } else if ch == ")" {
      if !queue.isEmpty && queue[0] == "(" {
        isValid = true
        queue.removeFirst()
      } else {
        isValid = false
        break
      }
    }
  }
  return queue.count == 0 && isValid
}